### ISSUE TITLE
SharePoint Framework SLA

### ISSUE THREAD

**Issue by wictorwilen** [SharePoint/sp-dev-docs#2399]
#### Category
- [X] Question
- [ ] Typo
- [ ] Bug
- [ ] Additional article idea

#### Question

What are the SLA's for SharePoint Framework?

We all know that the Office 365 support channels don't even bother supporting any customizations or SPFx. But what are the service level agreements for SharePoint Framework, what can we expect as developers and enterprises using SharePoint Framework as a business crucial component of their modern workplaces.

For instance:

* What are the official channel of requesting support?
  Is it this forum on Github? If so, how long can we expect an issue to be answered and eventually be resolved?
* How long will the different versions of SPFx be supported?
  How long will a major/minor/patch version for SPFx be supported? N-1 versions, 12 months, 4 years?
  There's been changes/removal in non-preview/beta code that prohibits upgrades to newer versions of the framework - how long can one stay on an older version? Removal of features has typically (in SharePoint history) been deprecation warnings and not point-blank nuke of functionality. Can we expect more of these?

*Asking for a friend ;)*

**Comment by nero120** [comment]
I'd be very interested to know this also. The amount of issues that the business I work for has had with production SPFX deployments, and the very difficult time we've had going through official channels (TAM, O365 support) means that this is the only place we can get any sort of suitable response.

However, this is not appropriate for an enterprise business and (along with the amount of issues we've experienced, i.e. not production ready) is making the development manager at my company _very_ nervous, to the point where he's considering pulling future development using SharePoint/SPFX.

**Comment by VesaJuvonen** [comment]
Completely understandable asks and let me try to clarify the situation in best possible ways.

- As SharePoint Dev support is not provided through the SharePoint Online support channels, we from the engineering have opened this issue list for reporting issues. This is to ensure that there's at least some channel for anyone to report any critical issues they might have with SharePoint development
- We triage typically incoming requests/questions/issues twice a week in the engineering side and route them to the right people. This is also when we prioritize the issue internally. Prioritization impacts the issue SLA, but it's not visible externally.
- If customer/partner has Premier Support agreement, that should be the default way to report anything critical, so that the support item has proper SLAs and escalation engineers handling the progress on the daily basis.
- We have recently increased investments on this support model, so you will see more changes and more actions and assistance also directly in here as we get things moving.

-- -- 

On the separate topic of things. SharePoint Framework is designed to be backward compatible and we guarantee that any GA/production deployed web part or extension will work in the SPO also in future. This means that also SPFx v1.0 web parts should work in SPO without any issues now and in upcoming years. If this would not be the case, it would be a bug for us to solve. 

--- 

Closing this for now as answered, but happy provide more details where suitable.
